src/FlightMap/Widgets/BatteryVisual.qml

```diff
@@ -0,0 +1,115 @@
+import QtQuick 2.12
+import QtQuick.Layouts 1.12
+import QGroundControl 1.0
+import QGroundControl.Controls 1.0
+import QGroundControl.ScreenTools 1.0
+import QGroundControl.Palette 1.0
+import QGroundControl.SettingsManager       1.0
+import QGroundControl.FactSystem            1.0
+import QGroundControl.FactControls          1.0
+
+Item {
+    id:           root
+    width:        50
+
+    property var _aviantSettings: QGroundControl.settingsManager.aviantSettings
+    property var batteryItem
+
+    Rectangle {
+        id:           batteryContainer
+        anchors.fill: parent
+        color:        "transparent"
+        border.color: qgcPal.text
+        radius:       width * 0.15
+        
+        FactPanelController { id: controller }
+        property bool vehicleReady: QGroundControl.multiVehicleManager.activeVehicle 
+                                    && QGroundControl.multiVehicleManager.activeVehicle.initialConnectComplete
+        property Fact batConsumed:           batteryItem ? batteryItem.mahConsumed : null
+        property Fact batCapacity:           vehicleReady ? controller.getParameterFact(-1, "BAT1_CAPACITY") : null
+        property Fact batCriticalThreshold:  vehicleReady ? controller.getParameterFact(-1, "BAT_CRIT_THR") : null
+        property Fact batEmergencyThreshold: vehicleReady ? controller.getParameterFact(-1, "BAT_EMERGEN_THR") : null
```

Comment:
Is this code re-run if vehicle is initally not ready but then becomes ready? Also I would check if this works when connecting between real vehicles, seems tied to multi-vehicle stuff.

Consider also getting the values from the controller to solve related issues